docs/synchronization.md

```diff
@@ -75,7 +75,7 @@ spec:
 ### Template-level Synchronization
 
 Template-level synchronization limits parallel execution of the template across workflows, if templates have the same synchronization reference.
-In this example, `acquire-lock` template has synchronization reference of `template` key which is configured as rate limit 2,
+In this example, `acquire-lock` template has synchronization reference of `template` key which is configured as limit 2,
 so two instances of templates will be executed at a given time: even multiple steps/tasks within workflow or different workflows referring to the same template.
```

Comment:
I found the use of the term "rate limit" confusing here but maybe my definition of "rate limit" is "limited"? :)